src/cms/models/languages/language.py

```diff
@@ -106,6 +106,18 @@ def translated_name(self):
         """
         return ugettext(self.english_name)
 
+    def get_source_language(self, region):
+        """
+        This returns the source language of this language in the given region
+
+        :return: The source language of the language
+        :rtype: ~cms.models.languages.language.Language
+        """
+        language_tree_node = self.language_tree_nodes.filter(region=region).first()
+        if language_tree_node and language_tree_node.parent:
```

Comment:
Why is the first item in the queryset expected to be the parent? Wouldn't it be necessary to iterate over all languages until the parent is found?

Reply:
`self` is an instance of the `Language` model, so `self.language_tree_nodes` is the queryset of all all instances of the `LanguageTreeNode` model which are mapped to this language (but across all regions). Since there cannot be two language tree nodes with the same language and the same region (because of [this constraint](https://github.com/Integreat/integreat-cms/blob/4e5560ddd0a2cd4708ae544b36358eb835bec41b/src/cms/models/languages/language_tree_node.py#L186-L191)), we know that the queryset `self.language_tree_nodes.filter(region=region)` can contain only one element (or none if the language doesn't exist in the requested region).
But yeah, I'll add a comment for better understanding.

Reply:
You're right. I was not thinking hard enough about the language tree. That makes totally sense.